cut v2.35.0-rc0 (#10543)
prometheus/prometheus: CHANGELOG.md

## 2.35.0-rc0 / 2022-04-07

* [CHANGE] TSDB: Delete *.tmp WAL files when Prometheus starts. #10317

> * [CHANGE] promtool: Add new flag `--lint`  for the commands `check rules` and `check config`. #10435

Reviewer: To explain why this is a CHANGE:

```suggestion
* [CHANGE] promtool: Add new flag `--lint` (enabled by default) for the commands `check rules` and `check config`, resulting in a new exit code (`3`) for linter errors. #10435
```

> * [ENHANCEMENT] promtool: Add new flag `--lint`  for the commands `check rules` and `check config`. #10435

Reviewer 2: Because we changed an exit code here, we should mark it as a `CHANGE` which I believe we did last time.

Author: Got it, I updated it to mark it as a `CHANGE`. I'm wondering if we should highlight that the exit code change. What do you think @LeviHarrison ?

Reviewer 2: The sole linter is enabled by default (as it had been previously), so I believe that changing the exit code it returns fits the criteria of a change `CHANGE`.

Reviewer: Ah ok, that makes sense then as a `CHANGE`, thanks for the clarification!

* [ENHANCEMENT] Prometheus is built with Go 1.18. #10501

> * [ENHANCEMENT] Log gomaxprocs messages. #10506

Reviewer 3: The actual feature is experimental support for gomaxprocs behind a feature flag - #10498

Author: ah I missed this one.
Thank you, Changelog is updated.

> * [FEATURE] Support for automatically set the variable `GOMAXPROCS` with the container CPU limit. Enable with the flag `--enable-feature=auto-gomaxprocs` #10498

Reviewer: ```suggestion
* [FEATURE] Support for automatically setting the variable `GOMAXPROCS` to the container CPU limit. Enable with the flag `--enable-feature=auto-gomaxprocs` #10498
```

## 2.35.0-rc0 / 2022-04-06

> * [CHANGE] Files *.tmp WAL are deleted when Prometheus is starting. #10317

Reviewer: Suggestion:

```suggestion
* [CHANGE] TSDB: Delete *.tmp WAL files when Prometheus starts. #10317
```

> * [FEATURE] ServiceDiscovery have now a plugin system. #10495

Reviewer: Hmmmm... as far as I can see this is not directly a user-visible feature, at least not in the sense that the binary that we release now has a user-visible plugin system. If I understand correctly, this is only to make it easier for people building Prometheus themselves to include/exclude SD methods. @roidelapluie What do you think about how to communicate this?

Author: Yeah indeed, but I was thinking since it's a nice changes in the Prometheus ecosystem, perhaps it will be interesting to highlight it in the release.

Reviewer: By our current changelog standards, we wouldn't mention it at all if it's not a direct user-visible change in the binaries we distribute. I think it's interesting in general for developers/builders (maybe as something to blog about it), but maybe not as part of these release notes?

Author: ah ok. In that case, yeah it's probably better to talk about it in a blog post or in a format @roidelapluie will find appropriate.

I removed it

> * [ENHANCEMENT] Add new flag `--lint`  for the commands `check rules` and `check config`. #10435

Reviewer: ```suggestion
* [ENHANCEMENT] promtool: Add new flag `--lint`  for the commands `check rules` and `check config`. #10435
```

* [ENHANCEMENT] Prometheus is built with Go 1.18. #10501
* [ENHANCEMENT] Log gomaxprocs messages. #10506
* [ENHANCEMENT] TSDB: more efficient sorting of postings read from WAL at startup. #10500

> * [ENHANCEMENT] Azure SD: Add metric to track Azure SD failure #10476

Reviewer: ```suggestion
* [ENHANCEMENT] Azure SD: Add metric to track Azure SD failures. #10476
```

Reviewer: As far as I can see, the PR also contains a small bugfix to the error handling around getting network interfaces?

* [ENHANCEMENT] Azure SD: Add an optional `resource_group` configuration. #10365

> * [ENHANCEMENT] Kubernetes SD: Support discovery.k8s.io/v1 EndpointSlice (previously only discovery.k8s.io/v1beta1 Endpoint slice was supported). #9570

Reviewer: ```suggestion
* [ENHANCEMENT] Kubernetes SD: Support `discovery.k8s.io/v1` `EndpointSlice` (previously only `discovery.k8s.io/v1beta1` `EndpointSlice` was supported). #9570
```

## 2.35.0-rc0 / 2022-04-08

* [CHANGE] TSDB: Delete *.tmp WAL files when Prometheus starts. #10317
* [CHANGE] promtool: Add new flag `--lint` (enabled by default) for the commands `check rules` and `check config`, resulting in a new exit code (`3`) for linter errors. #10435
* [FEATURE] Support for automatically setting the variable `GOMAXPROCS` to the container CPU limit. Enable with the flag `--enable-feature=auto-gomaxprocs` #10498

> * [FEATURE] Web: Add a new value: `all` for the query parameter `stats`. When used it will return the total number of samples read out and the previous statistic tracked per step in a range query. 

Reviewer 3: ```suggestion
* [FEATURE] PromQL: Extend statistics with total and peak number of samples in a query. Additionally, per-step statistics are available with  --enable-feature=promql-per-step-stats and using `stats=all` in the query API.
```
